kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseRuntimeDataServiceIntegrationTest.java

```diff
@@ -1411,6 +1416,115 @@ public void testGetCaseTasksAsStakeholder() throws Exception {
+        assertNotNull(task);
+        assertEquals(HELLO_1_TASK, task.getName());
+        assertEquals(USER_YODA, task.getActualOwner());
+
+        caseClient.triggerAdHocFragment(CONTAINER_ID, caseId, HELLO_2_TASK, Collections.EMPTY_MAP);
+
+        changeUser(USER_JOHN);
+        caseTasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseId, USER_JOHN, 0, 10);
+        assertNotNull(caseTasks);
+        assertEquals(1, caseTasks.size());
+
+        task = caseTasks.get(0);
+        assertNotNull(task);
+        assertEquals(HELLO_2_TASK, task.getName());
+        assertEquals(USER_JOHN, task.getActualOwner());
+
+        changeUser(USER_YODA);
+        caseClient.destroyCaseInstance(CONTAINER_ID, caseId);
+    }
+
+    @Test
+    public void testTriggerTaskIntoStage() throws Exception {
+        String caseClaimId = startCarInsuranceClaimCase(USER_YODA, USER_JOHN, USER_YODA);
+        assertNotNull(caseClaimId);
+
+        List<CaseStage> stages = caseClient.getStages(CONTAINER_ID, caseClaimId, true, 0, 10);
+        assertEquals(1, stages.size());
+        CaseStage stage = stages.iterator().next();
+        assertBuildClaimReportCaseStage(stage, "Active");
+
+        List<TaskSummary> tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        int countOfTaskBefore = tasks.size();
+
+        assertNotNull(stage.getIdentifier());
+        caseClient.triggerAdHocFragmentInStage(CONTAINER_ID, caseClaimId, stage.getIdentifier(), SUBMIT_POLICE_REPORT_TASK, Collections.EMPTY_MAP);
+
+        tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(countOfTaskBefore + 1, tasks.size());
+
+        TaskSummary task = tasks.get(0);
+        assertNotNull(task);
+        assertEquals(SUBMIT_POLICE_REPORT_TASK, task.getName());
+
+        caseClient.putCaseInstanceData(CONTAINER_ID, caseClaimId, "claimReportDone", Boolean.TRUE);
+        stages = caseClient.getStages(CONTAINER_ID, caseClaimId, false, 0, 10);
+        assertEquals(3, stages.size());
+        assertBuildClaimReportCaseStage(stages.get(0), "Completed");
+        assertClaimAssesmentCaseStage(stages.get(1), "Active");
+        assertEscalateRejectedClaimCaseStage(stages.get(2), "Available");
+
+        tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(1, tasks.size());
+        task = tasks.get(0);
+        assertNotEquals(SUBMIT_POLICE_REPORT_TASK, task.getName());
+
+        try {
+            caseClient.triggerAdHocFragmentInStage(CONTAINER_ID, caseClaimId, stage.getIdentifier(), SUBMIT_POLICE_REPORT_TASK, Collections.EMPTY_MAP);
+            fail("Task was triggered to completed stage");
+        } catch (KieServicesHttpException e) {
+            assertEquals(Integer.valueOf(400), e.getHttpCode());
```

Comment:
You can replace this by RestJmsSharedBaseIntegrationTest.assertClientException(...).